### DISCUSSION TITLE
alfa_bank

### DISCUSSION THREAD

**Discussion by oru13t** [Asebaie/Triple-I#4] [Q&A] [answered]
Почему используешь CatBoost+LightGBM, вместо, например, XGBoost и Optuna?

**Comment by Asebaie** [comment, marked as answer]
CatBoost лучше XGBoost в обработке категориальных признаков.
LightGBM лучше XGBoost по скорости обучения, инференса и эффективной работе с большим числом признаков.
Обе библиотеки будут лучше XGBoost, т.к. поддерживают sample_weight в регрессии, что позволяет напрямую оптимизировать под WMAE без кастомных loss-функций.
Optuna — это вообще отдельная история: она не заменяет ансамбль разных моделей, а лишь подбирает гиперпараметры. Плюсом Optuna не даёт преимущества в декорреляции моделей и не решает ограничения XGBoost в работе с грязными данными.
